### PR TITLE
Security_best-practices translated into Russian

### PR DESCRIPTION
**Pull request by ArduanovDanil**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | ❌

### CODE REVIEW

File: guide/ru/concept/security/best-practices.md
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:1]: Move file to `guide/ru/security/best-practices.md`

File: guide/ru/concept/security/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
+# Лучшие практики безопасности
+
+Ниже мы рассмотрим общие принципы безопасности и опишем, как избежать угроз при разработке приложений с использованием Yii . Большинство из этих принципов не являются уникальными только для Yii, но применимы к разработке веб-сайтов или программного обеспечения в целом, так что вы также найдете ссылки для дальнейшего чтения об общих идеях, лежащих в их основе.
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:3]:
```suggestion
Ниже мы рассмотрим общие принципы безопасности и опишем, как избежать угроз при разработке приложений с использованием Yii. Большинство из этих принципов не являются уникальными только для Yii, но применимы к разработке веб-сайтов или программного обеспечения в целом, так что вы также найдете ссылки для дальнейшего чтения об общих идеях, лежащих в их основе.
```

File: guide/ru/concept/security/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+## Основные принципы
+
+Независимо от того, какое приложение разрабатывается, существуют два основных принципа обеспечения безопасности:
+
+1. Фильтрация ввода.
+2. Экранирование вывода.
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:11]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+В контексте HTML вы должны экранировать `<`, `>` и похожие специальные символы.
+В контексте JavaScript или SQL это будет другой набор символов.
+Так как ручное экранирование чревато ошибками, Yii предоставляет различные утилиты для экранирования в различных контекстах.
+
+Дополнительная информация по теме:
+
+- <https://owasp.org/www-community/attacks/Command_Injection>
+- <https://owasp.org/www-community/attacks/Code_Injection>
+- <https://owasp.org/www-community/attacks/xss/>
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:46]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+## Как избежать SQL-иньекций
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:48]:
```suggestion
## Как избежать SQL-инъекций
```

File: guide/ru/concept/security/best-practices.md [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+SQL-иньекции происходят, когда текст запроса формируется склеиванием неэкранированных строк, как показано ниже:
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:50]:
```suggestion
SQL-инъекции происходят, когда текст запроса формируется склеиванием неэкранированных строк, как показано ниже:
```

File: guide/ru/concept/security/best-practices.md [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    }
+    
+    // ...
+}
+```
+
+Дополнительная информация по теме:
+
+- <https://owasp.org/www-community/attacks/SQL_Injection>
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:84]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+<?= \Yiisoft\Html\Html::encode($username) ?>
+```
+
+Если нужно вывести HTML, вам лучше воспользоваться [HtmlPurifier](http://htmlpurifier.org/).
+Обратите внимание, что обработка с помощью HtmlPurifier довольно тяжела, поэтому рассмотрите возможность использования кеширования.
+
+Дополнительная информация по теме:
+
+- <https://owasp.org/www-community/attacks/xss/>
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:109]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+2. Держите защиту CSRF в Yii включенной.
+
+Yii имеет защиту от CSRF в middleware `Yiisoft\Yii\Web\Middleware\Csrf`.
+Убедитесь, что он используется в вашем приложении.
+
+Дополнительная информация по теме:
+
+- <https://owasp.org/www-community/attacks/csrf>
+- <https://owasp.org/www-community/SameSite>
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:142]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+## Как избежать нежелательного доступа к файлам
+
+По умолчанию, webroot сервера указывает на каталог `public`, где лежит `index.php`. 
+В случае использования виртуального хостинга, это может быть недостижимо, в конечном итоге весь код, конфиги и логи могут оказаться в webroot сервера.
+
+Если это так, то нужно запретить доступ ко всему, кроме директории `web`. 
+Если на вашем хостинге такое невозможно, рассмотрите возможность смены хостинга.
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:151]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+
+## Как избежать вывода отладочной информации и инструментов в боевом окружении
+
+В режиме отладки, Yii отображает довольно подробные ошибки, которые полезны во время разработки.Однако, подробные ошибки удобны и для нападающего, так как могут раскрыть структуру базы данных, параметров конфигурации и части вашего кода.
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:155]:
```suggestion
В режиме отладки, Yii отображает довольно подробные ошибки, которые полезны во время разработки. Однако, подробные ошибки удобны и для нападающего, так как могут раскрыть структуру базы данных, параметров конфигурации и части вашего кода.
```

File: guide/ru/concept/security/best-practices.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+
+Вы никогда не должны оставлять Debug панель или Gii доступной для всех в боевом окружении. Это может быть использовано для получения информации о структуре базы данных или коде, может позволить заменить файлы, генерируемые Gii автоматически.
+
+Следует избегать включения в боевом окружении панели отладки, если только в этом нет острой необходимости. Она раскрывает всё приложение и детали конфигурации. Если вам всё-таки нужно запустить панель отладки, проверьте дважды, что доступ ограничен только вашими IP-адресами.
+
+Дополнительная информация по теме:
+
+- <https://owasp.org/www-project-.net/articles/Exception_Handling.md>
+- <https://owasp.org/www-pdf-archive/OWASP_Top_10_2007.pdf>
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:165]:
```suggestion
```

File: guide/ru/concept/security/best-practices.md [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+## Использование безопасного подключения через TLS
+
+Yii предоставляет функции, которые зависят от куки-файлов и/или сессий PHP. 
+Они могут быть уязвимыми, если ваше соединение скомпрометировано.
+
+Риск снижается, если приложение использует безопасное соединение через TLS (часто называемое как [SSL](https://en.wikipedia.org/wiki/Transport_Layer_Security)).
+
+Сегодня любой желающий может бесплатно получить SSL-сертификат и автоматически обновлять его благодаря [Let's Encrypt](https://letsencrypt.org/).
+
```
Comment (vjik) [inline guide/ru/concept/security/best-practices.md:175]:
```suggestion
```

**Review by vjik — CHANGES_REQUESTED** [review]

File: guide/ru/concept/security/best-practices.md
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md:1]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 1 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 2 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 3 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 6 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 7 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 8 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 5 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 9 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 10 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 11 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

File: guide/ru/concept/security/best-practices.md (hunk 12 above)
Comment (ArduanovDanil) [reply guide/ru/concept/security/best-practices.md]: Fixed

**Comment by vjik** [comment]
Great. Thanks 👍

**Pull request merged**